Sources/Basics/FileSystem/FileSystem+Extensions.swift

```diff
@@ -198,6 +198,18 @@ extension FileSystem {
     public func withLock<T>(on path: AbsolutePath, type: FileLock.LockType, _ body: () throws -> T) throws -> T {
         try self.withLock(on: path.underlying, type: type, body)
     }
+
+    public func capitalizeDriveLetterIfOnWindows(_ path: String) -> String {
+        #if !os(Windows)
+        return path
+        #endif
+        guard path.count > 2,
+            path.prefix(2).last == ":",
+            let drive = path.first,
+            drive >= "a" && drive <= "z"
+        else { return path }
+        return drive.uppercased() + path.suffix(path.count - 1)
```

Comment:
I think that I would rather see this spelt:

```swift
var iter = path.makeIterator()
guard iter.next().isLetter.isUppercase && iter.next() == ":" else { return path }
return "\(path.first.uppercased())\(path.dropFirst())"
```

I do think that @tomerd has a point on this being part of `AbsolutePath`.  I think that we want this normalised in all cases.

Reply:
"in all cases" means in all cases.  We want this to happen always when constructing a path, not just when constructing the contents for the SPM generated bundle.

Sinking this into t-s-c should effectively accomplish that I believe: https://github.com/apple/swift-tools-support-core/pull/422.